src/TaskLayout.ts

```diff
@@ -63,57 +63,55 @@ export class TaskLayout {
-        // isn't in the layout
-        const removeIf = (
-            taskComponents: TaskLayoutComponent[],
-            shouldRemove: boolean,
-            componentToRemove: TaskLayoutComponent,
-        ) => {
-            if (shouldRemove) {
-                this.taskListClasses.push(`tasks-layout-hide-${componentToRemove}`);
-                this.hiddenTaskLayoutComponents.push(componentToRemove);
-                return taskComponents.filter((element) => element != componentToRemove);
-            } else {
-                return taskComponents;
-            }
-        };
-        const markHiddenQueryComponent = (hidden: boolean, hiddenComponentName: string) => {
-            if (hidden) {
-                this.taskListClasses.push(`tasks-layout-hide-${hiddenComponentName}`);
-            }
-        };
+    private applyOptions() {
         // Remove components from the layout according to the task options. These represent the existing task options,
         // so some components (e.g. the description) are not here because there are no layout options to remove them.
-        let newComponents = this.shownTaskLayoutComponents;
-        newComponents = removeIf(newComponents, layoutOptions.hidePriority, 'priority');
-        newComponents = removeIf(newComponents, layoutOptions.hideRecurrenceRule, 'recurrenceRule');
-        newComponents = removeIf(newComponents, layoutOptions.hideCreatedDate, 'createdDate');
-        newComponents = removeIf(newComponents, layoutOptions.hideStartDate, 'startDate');
-        newComponents = removeIf(newComponents, layoutOptions.hideScheduledDate, 'scheduledDate');
-        newComponents = removeIf(newComponents, layoutOptions.hideDueDate, 'dueDate');
-        newComponents = removeIf(newComponents, layoutOptions.hideDoneDate, 'doneDate');
+        this.hideComponent(this.options.hidePriority, 'priority');
+        this.hideComponent(this.options.hideRecurrenceRule, 'recurrenceRule');
+        this.hideComponent(this.options.hideCreatedDate, 'createdDate');
+        this.hideComponent(this.options.hideStartDate, 'startDate');
+        this.hideComponent(this.options.hideScheduledDate, 'scheduledDate');
+        this.hideComponent(this.options.hideDueDate, 'dueDate');
+        this.hideComponent(this.options.hideDoneDate, 'doneDate');
+
+        this.generateHiddenClassForTaskList(this.options.hidePriority, 'priority');
+        this.generateHiddenClassForTaskList(this.options.hideRecurrenceRule, 'recurrenceRule');
+        this.generateHiddenClassForTaskList(this.options.hideCreatedDate, 'createdDate');
+        this.generateHiddenClassForTaskList(this.options.hideStartDate, 'startDate');
+        this.generateHiddenClassForTaskList(this.options.hideScheduledDate, 'scheduledDate');
+        this.generateHiddenClassForTaskList(this.options.hideDueDate, 'dueDate');
+        this.generateHiddenClassForTaskList(this.options.hideDoneDate, 'doneDate');
```

Comment:
Sorry but I am not fully comfortable with this change.

This seems like it is harder to maintain now than it was before, and potentially more error-prone. It's likely to generate future bugs.

I had to read carefully each pair of lines - 72 + 80, 72 + 81, to check that their args were exactly identical - and to count the total number of lines, to make sure there were none missing.

Why not have a single function that does both `hideComponent` and `generateHiddenClasses`?

Reply:
The problem here is that components are not treated in the same way. I hope that this will change one day (Like the next time a change in the layout will be needed maybe...). In this case there will be just one function indeed.

So for now I just wanted to split concerns here - manage the classes and the hiding/showing components with 2 arrays.

Maybe splitting the series of calls to two separate methods will be better? Or what would you prefer?
```js
this.hideComponents();
this.generateHiddenClassesForTasksList();
```

does this seem viable for you or you prefer something else?

Reply:
I'm really, really busy at the moment. Let's chat about this next time we pair.